### PR TITLE
Program security - Owner check lesson updated

### PR DESCRIPTION
**Pull request by 0xCipherCoder**
### Summary of Changes

- Updated code snippets with the latest anchor version
- Fixed content,
- Fixed grammar and styling
- Fixed as per guidelines

Fixes #
Unboxed PRs
Starter - https://github.com/Unboxed-Software/solana-owner-checks/pull/5
Solution - https://github.com/Unboxed-Software/solana-owner-checks/pull/6

### CODE REVIEW

File: content/courses/program-security/owner-checks.md [hunk 1]
```diff
@@ -3,10 +3,10 @@ title: Owner Checks
 objectives:
   - Explain the security risks associated with not performing appropriate owner
     checks
-  - Implement owner checks using long-form Rust
-  - Use Anchor’s `Account<'info, T>` wrapper and an account type to automate
+  - Implement owner checks using native Rust
+  - Use Anchor's `Account<'info, T>` wrapper and an account type to automate
```
Comment (mikemaccana) [inline content/courses/program-security/owner-checks.md:7]: Just swap these two (and do it everywhere, Anchor is default for onchain programs)

File: content/courses/program-security/owner-checks.md [hunk 2]
```diff
@@ -15,32 +15,34 @@ description:
...
-program from being used in an instruction.
+Owner checks are used to verify that an account passed into an instruction
+handler is owned by the expected program, preventing exploitation by accounts
+from different programs.
 
-As a refresher, the `AccountInfo` struct contains the following fields. An owner
-check refers to checking that the `owner` field in the `AccountInfo` matches an
-expected program ID.
+The `AccountInfo` struct contains several fields, including the `owner`, which
+represents the program that owns the account. Owner checks ensure that this
```
Comment (mikemaccana) [inline content/courses/program-security/owner-checks.md:44]:
```suggestion
represents the **program** that owns the account. Owner checks ensure that this
```

File: content/courses/program-security/owner-checks.md [hunk 3]
```diff
@@ -227,19 +202,18 @@ pub struct AdminConfig {
...
+### Use Anchor's `#[account(owner = <expr>)]` constraint
 
-In addition to the `Account` type, you can use an `owner` constraint. The
-`owner` constraint allows you to define the program that should own an account
-if it’s different from the currently executing one. This comes in handy if, for
-example, you are writing an instruction that expects an account to be a PDA
-derived from a different program. You can use the `seeds` and `bump` constraints
-and define the `owner` to properly derive and verify the address of the account
-passed in.
+In addition to the `Account` type, you can use the `owner` constraint to specify
```
Comment (mikemaccana) [inline content/courses/program-security/owner-checks.md:207]: Add a link to https://www.anchor-lang.com/docs/account-constraints (unfortunately Anchor doesn't support, er, anchors)

**Review by mikemaccana — APPROVED** [review]
Fine! Have some small changes,  please do them, but happy to mark this as the winner for owner-checks.

File: content/courses/program-security/owner-checks.md (hunk 3 above)
Comment (0xCipherCoder) [reply content/courses/program-security/owner-checks.md]: Updated anchor link as per comment.

**Comment by 0xCipherCoder** [comment]
> Fine! Have some small suggested changes, please do them, but happy to mark this as the winner for owner-checks.

Thanks! @mikemaccana  I have updated the content per the comment and resolved all of them. Also refactored tests in the starter and solution branches and updated the lesson content accordingly. Please review and let me know if any additional feedback.

**Comment by mikemaccana** [comment]
@0xCipherCoder can you change repo references to https://github.com/solana-developers/owner-checks/?

**Comment by 0xCipherCoder** [comment]
> @0xCipherCoder can you change repo references to https://github.com/solana-developers/owner-checks/?

@mikemaccana I have updated repo references. Please check.

**Comment by mikemaccana** [comment]
Thanks @0xCipherCoder !

**Pull request merged**